### PR TITLE
prompt for login_hint if it wasn't presented

### LINKED ISSUE #166: Portier Broker could prompt for login_hint
The client is currently required to send:
- `login_hint`
- ~~`nonce`~~

As an example, Dex does not send these, so is currently not working with Portier.

~~I don't see that either of those should be explicitly required. It seems that `state` is used for security validation most times anyway, and that `portier-broker` could have another page that prompts the user for entering their email instead of every OIDC client needing to implement a way to get `login_hint` from the user.~~

I think `login_hint` could be omitted and Portier could prompt for it.

### PR DESCRIPTION
**Pull request by colemickens**
Fixes #166.

Screenshot:
screenshot

### CODE REVIEW

**Comment by colemickens** [comment]
I've rebased this, and tests are passing.

File: src/handlers/auth.rs [hunk 1]
```diff
@@ -109,12 +115,39 @@ pub fn auth(ctx_handle: &ContextHandle) -> HandlerResult {
...
-    let login_hint = try_get_input_param!(params, "login_hint");
     if try_get_input_param!(params, "response_type") != "id_token" {
         return Box::new(future::err(BrokerError::Input(
             "unsupported response_type, only id_token is supported".to_owned())));
     }
 
+    let login_hint = try_get_input_param!(params, "login_hint", "".to_string());
+    if login_hint == "" {
+        let data = mustache::MapBuilder::new()
+            // TODO: catalog/localization?
```
Comment (stephank) [inline src/handlers/auth.rs]: Is this something you'd like to try, or should we take a look at it?

Comment (colemickens) [reply src/handlers/auth.rs]: I went ahead and implemented it, following the pattern of other localized strings.

I'm not qualified to localize for other languages, though, unfortunately.

Tests are green as well. :slightly_smiling_face:

**Comment by colemickens** [comment]
And the PR for the spec: https://github.com/portier/portier.github.io/pull/37

**Comment by onli** [comment]
This can be merged, no?

**Comment by onli** [comment]
Thank you! :)

**Pull request merged**